c/libeulabeia/include/eulabeia/types.h

```diff
@@ -0,0 +1,272 @@
+#ifndef __EULABEIA_DATA_H
+#define __EULABEIA_DATA_H
+/**
+ * @brief defines the scan result states.
+ *
+ * The first parameter is used to generate enum eulabeia_scan_result_state and
+ * the second parameter is used to translate from and to the json
+ * representation.
+ *
+ * To add a new state add it here and NOT within enum
+ * eulabeia_scan_result_state.
+ *
+ */
+#define EULABEIA_SCAN_RESULT_STATES                                            \
```

Comment:
I am not quite sure what you mean with result state. Is it the state of a scan?

Reply:
yes

Reply:
then why is result contained in the name?

Reply:
from a client point of view it is all about results so not such much about the scan itself; since we already have an aggregate named scan which is more likely to be used then the status of a particular running scan except for finish result_states I thought result_states is clearer than just scan_states.

Reply:
Also the struct EulabeiaScanProgress will contain results to allow periodically flushing results into db which also played into the name EULABEIA_SCAN_RESULT_STATES.

Reply:
Anyway going to rename it to EULABEIA_SCAN_STATE.